pkg/maps/configmap/configmap.go

```diff
@@ -27,5 +27,187 @@ import (
+	}
+
+	binMap = map[uint]string{
+		0: "POLICY_INGRESS",
+		1: "POLICY_EGRESS",
+	}
+
+	// configKey is the key in each endpoint's map for its configuration.
+	configKey = Key{Bits: 0}
 )
+
+// Flags is a set of endpoint configuration flags interpreted by BPF code.
+//
+// Must be in sync with the enum ep_cfg_flag in <bpf/lib/eps.h>
+type Flags uint32
+
+// String converts the specified flags into a human-readable form.
+func (f Flags) String() string {
+	var buffer bytes.Buffer
+	for i := uint(0); i < 32; i++ {
+		bitVal := int(f & (1 << i))
+		if bitVal > 0 {
+			if flag, ok := flagsToString[bitVal]; ok {
+				buffer.WriteString(flag)
+				buffer.WriteString(",")
+			}
+		} else if bitVal == 0 {
+			if flag, ok := binMap[i]; ok {
+				buffer.WriteString(flag)
+				buffer.WriteString(",")
+			}
+		}
+	}
+
+	return buffer.String()
+}
+
+// Key is the key used to index into the config map for an endpoint.
+//
+// Must be in sync with the key of CONFIG_MAP in <bpf/lib/maps.h>
+type Key struct {
+	Bits uint32
+}
+
+func (k Key) String() string {
+	return fmt.Sprintf("%d", k)
+}
+
+func (k Key) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(&k) }
+
+// NewValue returns a new empty instance of the structure representing the BPF
+// map value
+func (k Key) NewValue() bpf.MapValue { return &EndpointConfig{} }
+
+func (cfg EndpointConfig) String() string {
+	return cfg.Flags.String()
+}
+
+// EndpointInfo represents the value of the endpoints BPF map.
+//
+// Must be in sync with struct config_value in <bpf/lib/common.h>
+type EndpointConfig struct {
+	Flags Flags
+}
+
+// GetValuePtr returns the unsafe pointer to the BPF value
+func (c EndpointConfig) GetValuePtr() unsafe.Pointer { return unsafe.Pointer(&c) }
+
+// endpoint provides access to the properties of an endpoint that are relevant
+// for configuring the BPF program for the endpoint.
+type endpoint interface {
+	GetIngressPolicyEnabledLocked() bool
+	GetEgressPolicyEnabledLocked() bool
+}
+
+// EndpointConfigMap is a map type for interfacing with endpoint BPF config.
+type EndpointConfigMap struct {
+	Map  *bpf.Map
+	path string
+	Fd   int
+}
+
+// GetConfig creates a EndpointConfig structure using the endpoint's
+// configuration.
+func GetConfig(e endpoint) *EndpointConfig {
+	value := EndpointConfig{}
+	if !e.GetIngressPolicyEnabledLocked() {
+		value.Flags |= SkipPolicyIngress
+	}
+	if !e.GetEgressPolicyEnabledLocked() {
+		value.Flags |= SkipPolicyEgress
+	}
+
+	return &value
+}
+
+// Sync pushes the configuration options from the specified endpoint into the
+// configuration map.
+func (m *EndpointConfigMap) Update(value *EndpointConfig) error {
+	return m.Map.Update(configKey, value)
+}
+
+// OpenMap attempts to open or create a BPF config map at the specified path.
+// On success, it returns a map and whether the map was newly created, or
+// otherwise an error.
+func OpenMap(path string) (*EndpointConfigMap, bool, error) {
+
+	newMap := bpf.NewMap("cilium_lb4_services",
```

Comment:
Is this supposed to be the right name? :thinking:

Reply:
This function is dead code, sorry about that. I'll remove it in the next upload.